src/Components/Aspire.Elastic.Clients.Elasticsearch/AspireElasticClientsElasticsearchExtensions.cs

```diff
@@ -0,0 +1,149 @@
+        Action<ElasticsearchClientSettings>? configureClientSettings = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+        builder.AddElasticsearchClient(
+            $"{DefaultConfigSectionName}:{name}",
+            configureSettings,
+            configureClientSettings,
+            connectionName: name,
+            serviceKey: name);
+    }
+
+    private static void AddElasticsearchClient(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<ElasticClientsElasticsearchSettings>? configureSettings,
+        Action<ElasticsearchClientSettings>? configureClientSettings,
+        string connectionName,
+        object? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var configSection = builder.Configuration.GetSection(configurationSectionName);
+
+        ElasticClientsElasticsearchSettings settings = new();
+        configSection.Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ConnectionString = connectionString;
+        }
+
+        configureSettings?.Invoke(settings);
+
+        var elasticsearchClientSettings = CreateElasticsearchClientSettings(settings, connectionName, configurationSectionName);
+
+        configureClientSettings?.Invoke(elasticsearchClientSettings);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddSingleton<ElasticsearchClient>(sp => new ElasticsearchClient(elasticsearchClientSettings));
+        }
+        else
+        {
+            builder.Services.AddKeyedSingleton<ElasticsearchClient>(serviceKey, (sp, key) => new ElasticsearchClient(elasticsearchClientSettings));
+        }
+
+        if (!settings.DisableTracing)
+        {
+            builder.Services
+                .AddOpenTelemetry()
+                .WithTracing(tracer => tracer.AddSource(ActivityNameSource));
+        }
+
+        if (!settings.DisableHealthChecks)
+        {
+            var healthCheckName = serviceKey is null ? "Elastic.Clients.Elasticsearch" : $"Elastic.Clients.Elasticsearch_{connectionName}";
+
+            builder.TryAddHealthCheck(new HealthCheckRegistration(
+                healthCheckName,
+                sp => new ElasticsearchHealthCheck(serviceKey is null ?
+                    sp.GetRequiredService<ElasticsearchClient>() :
+                    sp.GetRequiredKeyedService<ElasticsearchClient>(serviceKey)),
+                failureStatus: null,
+                tags: null,
+                timeout: settings.HealthCheckTimeout > 0 ? TimeSpan.FromMilliseconds(settings.HealthCheckTimeout.Value) : null
+                ));
+        }
+    }
+
+    private static ElasticsearchClientSettings CreateElasticsearchClientSettings(
+        ElasticClientsElasticsearchSettings settings,
+        string connectionName,
+        string configurationSectionName)
+    {
+        if (settings.UseCloud)
```

Comment:
This `UseCloud` configuration feels superfluous perhaps? Setting `settings.Cloud` to anything but `null` signals the same intent.